src/dune_engine/process.ml

```diff
@@ -801,20 +801,23 @@ let run_internal ?dir ?(stdout_to = Io.stdout) ?(stderr_to = Io.stderr)
         | `Capture fn ->
           swallow_on_success_if_requested fn actual_stderr stderr_on_success
       in
-      let output = stdout ^ stderr in
-      Log.command ~command_line ~output ~exit_status:process_info.status;
-      let res =
-        match (display.verbosity, exit_status', output) with
-        | Quiet, Ok n, "" -> n (* Optimisation for the common case *)
-        | Verbose, _, _ ->
-          Handle_exit_status.verbose exit_status' ~id ~metadata ~dir
-            ~command_line:fancy_command_line ~output
-        | _ ->
-          Handle_exit_status.non_verbose exit_status' ~prog:prog_str ~dir
-            ~command_line ~output ~metadata ~verbosity:display.verbosity
-            ~has_unexpected_stdout ~has_unexpected_stderr
-      in
-      (res, times))
+      if Fiber.Cancel.fired cancel then
+        raise Dune_util.Report_error.Already_reported
```

Comment:
I find this line confusing, perhaps due to the ambiguous name `Already_reported`.

Here you are not reporting an error that wasn't reported before. Maybe some other error was "already reported" but not the one that we have here. 

Can we make the code/naming somehow less confusing?

Reply:
And, by the way, what if the job actually succeeded? Basically, the code seems to do the opposite of what you wrote in the PR description.

Reply:
> And, by the way, what if the job actually succeeded? Basically, the code seems to do the opposite of what you wrote in the PR description.

Could you clarify this a bit?

So what if the job succeeded? if the build job it belonged to was cancelled, we aren't interested in its output.

Reply:
I think I basically don't understand what your second bullet point means. At least, I don't see any change that corresponds to it?

Reply:
Oh, sorry, I think I get it now. Ignore me, I'm slow.

Reply:
Actually my description was hastily written. I'll update it and the commit message.

Reply:
So, this `if` that you added achieves two things at the same time and it wasn't obviuos to me. I suggest adding a comment that clarifies that.

Reply:
> Can we make the code/naming somehow less confusing?

I suppose we abuse this name in other places as well. For example, we treat user requested shutdowns initiated with a signal as `Already_reported` as well. I'm at a loss for a better name right now, but I'll leave a comment to clarify.